### PR TITLE
VOTE- 3160 touch points mobile menu

### PR DESCRIPTION
**Pull request by SamiraMSadat**
## Jira ticket

https://cm-jira.usa.gov/browse/VOTE-3160

## Description

Add a button to the touch points form at the bottom of the mobile menu.

### Post-deploy steps

1. lando retune
2. cd into theme, npm run build

### QA/Testing instructions

1. View the site in mobile dimensions and confirm the button matches the design.
2. Confirm the button displays for multiple screen sizes in mobile and tablet. Note the larger screen sizes place the button start aligned instead of centered.
3. Switch the Spanish and confirm the button no longer displays in the menu.
4. Switch to desktop and confirm the modal opens when the floating touch points button is clicked

## Checklist for the Developer

- [x] A link to the JIRA ticket has been included above.
- [x] No merge conflicts exist with the target branch.
- [x] Automated tests have passed on this PR.
- [x] A reviewer has been designated.
- [x] Deployment and testing steps have been documented above, if applicable.

## Checklist for the Peer Reviewers

- [x] The file changes are relevant to the task objective.
- [x] Code is readable and includes appropriate commenting.
- [x] Code standards and best practices are followed.
- [x] QA/Test steps were successfully completed, if applicable.
- [x] Applicable logs are free of errors.

### CODE REVIEW

**Review by clmedders — CHANGES_REQUESTED** [review]
The module button is showing on desktop 
Screenshot 2024-12-13 at 9 47 53 AM

File: web/themes/custom/votegov/templates/navigation/menu--main--votegov-mainnavigation.html.twig [hunk 1]
```diff
@@ -36,6 +36,12 @@ We call a macro which calls itself to render the full tree.
   <div class="usa-nav__secondary">
     {{ drupal_entity('block_content', 2) }}
   </div>
+  {% if language == 'en' %}
+      <div class="usa-nav__secondary">
+        <button class="usa-button" id="vote-touchpoints--button" title="{{ 'How can we improve vote.gov?' | t }}" data-test="touchpointsBtn">{{ 'How can we improve vote.gov?' | t }}</button>
+        <script src="https://touchpoints.app.cloud.gov/touchpoints/c169d3b2.js" async></script>
+      </div>
+  {% endif %}
```
Comment (clmedders) [inline web/themes/custom/votegov/templates/navigation/menu--main--votegov-mainnavigation.html.twig:44]: This needs to be in the div above to just show on mobile

File: web/themes/custom/votegov/templates/navigation/menu--main--votegov-mainnavigation.html.twig [hunk 2]
```diff
@@ -36,6 +36,12 @@ We call a macro which calls itself to render the full tree.
   <div class="usa-nav__secondary">
     {{ drupal_entity('block_content', 2) }}
   </div>
+  {% if language == 'en' %}
+      <div class="usa-nav__secondary">
```
Comment (clmedders) [inline web/themes/custom/votegov/templates/navigation/menu--main--votegov-mainnavigation.html.twig:40]: i dont think we need this div of nav__ secondary. when i removed the div it fixed the spacing issue

File: web/themes/custom/votegov/templates/navigation/menu--main--votegov-mainnavigation.html.twig [hunk 3]
```diff
@@ -35,6 +35,10 @@ We call a macro which calls itself to render the full tree.
   {{ menus.menu_links(items, attributes, 0, 'basic-mega-nav-section-') }}
   <div class="usa-nav__secondary">
     {{ drupal_entity('block_content', 2) }}
+     {% if language == 'en' %}
+        <button class="usa-button" id="vote-touchpoints--button" title="{{ 'How can we improve vote.gov?' | t }}" data-test="touchpointsBtn">{{ 'How can we improve vote.gov?' | t }}</button>
+        <script src="https://touchpoints.app.cloud.gov/touchpoints/c169d3b2.js" async></script>
```
Comment (rayestrada) [inline web/themes/custom/votegov/templates/navigation/menu--main--votegov-mainnavigation.html.twig:40]: Is this script tag loaded twice on the page?

Comment (SamiraMSadat) [reply web/themes/custom/votegov/templates/navigation/menu--main--votegov-mainnavigation.html.twig]: This is now only loaded once on the page.

File: web/themes/custom/votegov/templates/navigation/menu--main--votegov-mainnavigation.html.twig [hunk 4]
```diff
@@ -35,6 +35,10 @@ We call a macro which calls itself to render the full tree.
   {{ menus.menu_links(items, attributes, 0, 'basic-mega-nav-section-') }}
   <div class="usa-nav__secondary">
     {{ drupal_entity('block_content', 2) }}
+     {% if language == 'en' %}
+        <button class="usa-button" id="vote-touchpoints-mobile--button" title="{{ 'How can we improve vote.gov?' | t }}" data-test="touchpointsBtn"
+         onclick="document.getElementById('vote-touchpoints--button').click()">{{ 'How can we improve vote.gov?' | t }}</button>
```
Comment (rayestrada) [inline web/themes/custom/votegov/templates/navigation/menu--main--votegov-mainnavigation.html.twig:40]: It is not recommended to have onclick functions in the dom. Can you move this to a js file and load it with the button?

**Review by rayestrada — CHANGES_REQUESTED** [review]

**Review by rayestrada — APPROVED** [review]

**Pull request merged**